moz-webgpu-cts/src/main.rs

```diff
@@ -125,6 +190,474 @@ fn run(cli: Cli) -> ExitCode {
+                            MaybeDisabled::Disabled
+                        } else {
+                            MaybeDisabled::Enabled(expectations)
+                        };
+                        if let Some(_old) =
+                            recorded_subtest_outcomes.metadata.replace(maybe_disabled)
+                        {
+                            if !reported_dupe_already {
+                                freak_out_do_nothing(&lazy_format!(
+                                    "duplicate subtest in {test_path:?} named {subtest_name:?}, discarding previous entries with this and further dupes"
+                                ));
+                            }
+                        }
+                    }
+                }
+            }
+
+            log::info!("gathering reported test outcomes for reconciliation with metadata…");
+
+            let (exec_reports_sender, exec_reports_receiver) = channel();
+            exec_report_paths
+                .into_par_iter()
+                .for_each_with(exec_reports_sender, |sender, path| {
+                    let res = fs::File::open(&path)
+                        .map(BufReader::new)
+                        .map_err(Report::msg)
+                        .wrap_err("failed to open file")
+                        .and_then(|reader| {
+                            serde_json::from_reader::<_, ExecutionReport>(reader)
+                                .into_diagnostic()
+                                .wrap_err("failed to parse JSON")
+                        })
+                        .wrap_err_with(|| {
+                            format!(
+                                "failed to read WPT execution report from {}",
+                                path.display()
+                            )
+                        })
+                        .map(|parsed| (path, parsed))
+                        .map_err(|e| {
+                            log::error!("{e:?}");
+                            AlreadyReportedToCommandline
+                        });
+                    let _ = sender.send(res);
+                });
+
+            for res in exec_reports_receiver {
+                let (_path, exec_report) = match res {
+                    Ok(ok) => ok,
+                    Err(AlreadyReportedToCommandline) => return ExitCode::FAILURE,
+                };
+
+                let ExecutionReport {
+                    run_info:
+                        RunInfo {
+                            platform,
+                            build_profile,
+                        },
+                    entries,
+                } = exec_report;
+
+                for entry in entries {
+                    let TestExecutionEntry { test_name, result } = entry;
+
+                    let test_path = TestPath::from_execution_report(&test_name).unwrap();
+                    // TODO: ignore things outside of the WPT path(s) we care about
```

Comment:
This can be controlled by users not specifying reports that contain things irrelevant to WebGPU, which is pretty easy ATOW.

- [ ] Let's punt this to a follow-up issue.

Reply:
https://github.com/ErichDonGubler/moz-webgpu-cts/issues/51

Reply:
c20b45adff7e2470e72080954a8b3bd266d4891f